### PR TITLE
Eqn editor derivatives

### PR DESCRIPTION
**Pull request by Morpheu5**
Introduces derivatives. Needs the backend checker to support `diff` rather than `Derivative`. Otherwise ready. Needs review and thorough testing.

### CODE REVIEW

File: app/js/app/directives/equation_editor/EquationEditor.js [hunk 1]
```diff
@@ -771,6 +782,123 @@ define(function(require) {
                     return result;
                 };
 
+                var derivativeFunctions = function(availableDerivatives) {
+                    // TODO Possibly remove the menu if no derivatives are available
+                    var result = [];
+                    var userIsPrivileged = _.includes(['ADMIN', 'CONTENT_EDITOR', 'EVENT_MANAGER'], scope.user.role);
+
+                    if(userIsPrivileged) {
```
Comment (jsharkey13) [inline app/js/app/directives/equation_editor/EquationEditor.js]: You tried _so_ hard . . .

File: app/ts/inequality/Symbol.ts [hunk 2]
```diff
@@ -28,19 +28,29 @@ export
...
+        this.docksTo = ['relation', 'operator', 'exponent', 'symbol_subscript', 'symbol', 'operator_brackets', 'differential_argument'];
     }
 
 	/**
 	 * Generates all the docking points in one go and stores them in this.dockingPoints.
 	 * A Symbol has three docking points:
 	 *
 	 * - _right_: Binary operation (addition, subtraction), Symbol (multiplication)
-	 * - _superscript_: Exponent
+	 * - _superscript_: Exponent (unless it's the argument of a Differential (heh...)
```
Comment (jsharkey13) [inline app/ts/inequality/Symbol.ts]: Not enough closing brackets on this informative comment :P

File: app/ts/inequality/Symbol.ts [hunk 3]
```diff
@@ -203,24 +216,26 @@ export
...
 
-        if ("superscript" in boxes) {
-            child_width = docking_superscript.child.boundingBox().w;
-            child_height = docking_superscript.child.boundingBox().h;
-            docking_superscript.child.position.x = (parent_width / 2 + child_width / 2);
-            docking_superscript.child.position.y = -0.8 * (parent_height / 2 + child_height / 2);
-        } else {
-            docking_superscript.position.x = (parent_width == this.boundingBox().w) ? (parent_width / 2 + this.scale * 20) : (parent_width - this.boundingBox().w / 2 + this.scale * 20);
-            docking_superscript.position.y = -this.scale * this.s.mBox.h;
+        if(sonOfADifferential) {
```
Comment (jsharkey13) [inline app/ts/inequality/Symbol.ts]: And here!

File: app/ts/inequality/Symbol.ts (hunk 2 above)
Comment (Morpheu5) [inline app/ts/inequality/Symbol.ts]: I hate you!

**Comment by Morpheu5** [comment]
I'll have to take a look at this on Monday, then.

**Pull request merged**